metricflow/dataflow/builder/node_evaluator.py

```diff
@@ -112,7 +112,7 @@ class NodeEvaluatorForLinkableInstances(Generic[SourceDataSetT]):
 
     def __init__(
         self,
-        semantic_model_semantics: SemanticModelSemanticsAccessor,
+        semantic_model_semantics: AbstractSemanticModelAccessor,
```

Comment:
Ick. Can we call the abstract class SemanticModelAccessor and then have the concrete class use a modified name? This isn't terribly important so it's fine to say no and merge when ready. :)

Eventually we should just fix the API - the only reason this stuff exists is because there are weird methods in the original objects that exist only to facilitate runtime injection of what we now call SemanticModel objects.